tests/test_declaration.py

```diff
@@ -10,492 +20,543 @@
-        node.style["thing_bottom"] = 10
-        self.assertEqual(node.style["thing_bottom"], 10)
-        del node.style["thing_bottom"]
-        self.assertEqual(node.style["thing_bottom"], 0)
-
-        # Clearing a valid property isn't an error
-        del node.style["thing_bottom"]
-        self.assertEqual(node.style["thing_bottom"], 0)
-
-        # Non-existent properties raise KeyError
-        with self.assertRaises(KeyError):
-            node.style["no-such-property"] = "no-such-value"
-
-        with self.assertRaises(KeyError):
-            node.style["no-such-property"]
-
-        with self.assertRaises(KeyError):
-            del node.style["no-such-property"]
+    # Some properties with explicit initial values
+    explicit_const: str | int = validated_property(
+        choices=VALUE_CHOICES, initial=VALUE1
+    )
+    explicit_value: str | int = validated_property(choices=VALUE_CHOICES, initial=0)
+    explicit_none: str | int | None = validated_property(
+        choices=VALUE_CHOICES, initial=None
+    )
+
+    # A property with an implicit default value.
+    # This usually means the default is platform specific.
+    implicit: str | int | None = validated_property(choices=DEFAULT_VALUE_CHOICES)
+
+    # A set of directional properties
+    thing: str | int = directional_property("thing{}", choices=VALUE_CHOICES, initial=0)
+    thing_top: str | int
+    thing_right: str | int
+    thing_bottom: str | int
+    thing_left: str | int
+
+
+with catch_warnings():
+    filterwarnings("ignore", category=DeprecationWarning)
+
+    @mock_apply
+    class DeprecatedStyle(BaseStyle):
+        pass
+
+    # Some properties with explicit initial values
+    DeprecatedStyle.validated_property(
+        "explicit_const", choices=VALUE_CHOICES, initial=VALUE1
+    )
+    DeprecatedStyle.validated_property(
+        "explicit_value", choices=VALUE_CHOICES, initial=0
+    )
+    DeprecatedStyle.validated_property(
+        "explicit_none", choices=VALUE_CHOICES, initial=None
+    )
+
+    # A property with an implicit default value.
+    # This usually means the default is platform specific.
+    DeprecatedStyle.validated_property("implicit", choices=DEFAULT_VALUE_CHOICES)
+
+    # A set of directional properties
+    DeprecatedStyle.validated_property("thing_top", choices=VALUE_CHOICES, initial=0)
+    DeprecatedStyle.validated_property("thing_right", choices=VALUE_CHOICES, initial=0)
+    DeprecatedStyle.validated_property("thing_bottom", choices=VALUE_CHOICES, initial=0)
+    DeprecatedStyle.validated_property("thing_left", choices=VALUE_CHOICES, initial=0)
+    DeprecatedStyle.directional_property("thing%s")
+
+
+def test_invalid_style():
+    with pytest.raises(ValueError):
+        # Define a style that has an invalid initial value on a validated property
+        class BadStyle(BaseStyle):
+            value = validated_property(choices=VALUE_CHOICES, initial="something")
```

Comment:
Why doesn't this require a `@dataclass` decoration to trigger the error? Or, rather - is the class actually needed here? Isn't the issue just instantiating `validated_property()`? And isn't there a parallel non-dataclass version that needs testing?

Reply:
It was in a style originally, but yeah there's no reason it needs to be there, I can make it just the property.